sdk/tables/azure-data-tables/azure/data/tables/aio/_table_service_client_async.py

```diff
@@ -196,9 +212,8 @@ async def delete_table(
         :return: None
         :rtype: ~None
          """
-        _validate_table_name(table_name)
-
-        await self._client.table.delete(table=table_name, **kwargs)
+        table = self.get_table_client(table_name=table_name)
```

Comment:
we're losing checks to validate the table name

Reply:
The validation is done [here](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/tables/azure-data-tables/azure/data/tables/_table_client_base.py#L43) in the base client when it's attempted to instantiate, it would throw the same error.